extensions/amp-subscriptions/0.1/amp-subscriptions.js

```diff
@@ -257,6 +273,56 @@ export class SubscriptionService {
     return this;
   }
 
+  /**
+   * Delegates authentication to viewer
+   */
+  delegateAuthToViewer_() {
+    const serviceIds = ['local'];
+    const currentProductId = /** @type {string} */ (user().assert(
+        this.pageConfig_.getProductId(),
+        'Product id is null'
+    ));
+    const publicationId = /** @type {string} */ (user().assert(
+        this.pageConfig_.getPublicationId(),
+        'Publication id is null'
+    ));
+    const origin = getWinOrigin(this.ampdoc_.win);
+    this.platformStore_ = new PlatformStore(serviceIds);
+    this.viewer_.sendMessageAwaitResponse('auth', dict({
+      'publicationId': publicationId,
+      'productId': currentProductId,
+      'origin': origin,
+    })).then(entitlementData => {
+      const authData = (entitlementData || {})['authorization'];
+      if (!authData) {
+        return this.platformStore_.resolveEntitlement('local',
+            Entitlement.empty('local'));
+      }
+      const decodedData = this.jwtHelper_.decode(authData);
+      user().assert(decodedData['aud'] == origin, 'Bad Audience');
```

Comment:
Let's phrase it as `The mismatching "aud" field: ${docdedData['aud']}`

Reply:
Done